Source/Diagnostics/ComputeDiagFunctors/ParticleReductionFunctor.cpp

```diff
@@ -69,23 +69,7 @@ ParticleReductionFunctor::operator() (amrex::MultiFab& mf_dst, const int dcomp,
                 get_particle_position(p, xw, yw, zw);
 
                 // Get position in AMReX convention to calculate corresponding index.
-                // Ideally this will be replaced with the AMReX NGP interpolator
-                // Always do x direction. No RZ case because it's not implemented, and code
-                // will have aborted
-                int ii = 0, jj = 0, kk = 0;
-                const amrex::ParticleReal x = p.pos(0);
-                const amrex::Real lx = (x - plo[0]) * dxi[0];
-                ii = static_cast<int>(amrex::Math::floor(lx));
-#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_3D) || defined(WARPX_DIM_RZ)
-                const amrex::ParticleReal y = p.pos(1);
-                const amrex::Real ly = (y - plo[1]) * dxi[1];
-                jj = static_cast<int>(amrex::Math::floor(ly));
-#endif
-#if defined(WARPX_DIM_3D)
-                const amrex::ParticleReal z = p.pos(2);
-                const amrex::Real lz = (z - plo[2]) * dxi[2];
-                kk = static_cast<int>(amrex::Math::floor(lz));
-#endif
+                const auto [ii, jj, kk] = amrex::getParticleCell(p, plo, dxi).dim3();
```

Comment:
Ah, there is another branch, `m_do_average`, below that can use the same functionality.

Reply:
Follow-up in #5557